cldfbench_steinertthrelkeldmodals.py

```diff
@@ -73,32 +97,31 @@ def cmd_makecldf(self, args):
                         Value=modal,
                     )
                 )
-                modal_id += 1
                 unit_obs_id = 0
                 for can, rrrows in itertools.groupby(rrows, lambda r: r["can"]):
                     for row in rrrows:
                         unit_obs_id += 1
                         test_dict = dict(
                             ID=f"{modal_id}-{unit_obs_id}",
-                            Language_ID=lid,
                             Parameter_ID="modal",
-                            Value=modal,
+                            Value=str(modal_id),
                             UnitParameter_ID=f"{row['force']}.{row['flavor']}",
                             UnitValue="can" if can == "1" else "cannot",
                         )
                         force_flavor_pairs.add((row["force"], row["flavor"]))
                         args.writer.objects["unit-values.csv"].append(test_dict)
+                modal_id += 1
 
         for idx, pair in enumerate(force_flavor_pairs):
             # TODO: refactor naming of pairs
             args.writer.objects["unit-parameters.csv"].append(
-                dict(ID=idx, Name=f"{pair[0]}.{pair[1]}", force=pair[0], flavor=pair[1])
+                dict(ID=f"{pair[0]}.{pair[1]}", Name=f"{pair[0]}.{pair[1]}", force=pair[0], flavor=pair[1])
             )
 
-        forces = set(pair[0] for pair in force_flavor_pairs)
+        forces = sorted(set(pair[0] for pair in force_flavor_pairs))
```

Comment:
@shanest To make CLDF creation idempotent - and thus create meaningful, minimal diffs for changed data - I sort every `set` or `dict` explicitly before iterating over it to add data.

Reply:
Makes total sense!